### ISSUE TITLE
[YSQL] psycopg2.DatabaseError instance on serialization failure exception not of expected errors.SerializationFailure subclass

### ISSUE THREAD

**Issue by bllewell** [yugabyte/yugabyte-db#1567]
Using YB **Version 1.2.10.0** runing on **macOS Mojave 10.14.5**.

Here’s a snippet from a working Python test program:

  ```
try:
    cur.execute(statement)
  except (Exception, psycopg2.DatabaseError) as error:
    session.rollback()
    print("pgcode: " + str(error.pgcode))

    if isinstance(error, psycopg2.errors.SerializationFailure):
      print("got instance of the SerializationFailure subclass")

    elif isinstance(error, psycopg2.errors.UniqueViolation):
      print("got instance of the UniqueViolation subclass")

    else:
      raise error
```
When `statement` is an `insert` that would violate a unique constraint, I get this output in both YB and PosgreSQL:
```
pgcode: 23505
got instance of the UniqueViolation subclass
```
I assume that I'd see corresponding behavior for other errors. But when `statement` is anything that would cause a serialization failure, I get this output in YB:
```
pgcode: XX000
```
Notice that the print that would show this:
```
got instance of the errors.SerializationFailure subclass
```
is *not* output:

In other words, YB doesn’t construct an `errors.SerializationFailure` instance.

When the same test is run in a PostgreSQL database, I get this on a serialization error:
```
pgcode: 40001
got instance of the SerializationFailure subclass
```

**Comment by kmuthukk** [comment]
@mbautin - also assigning this one to you as discussed.

**Comment by ndeodhar** [comment]
Fixed as a result of #4114
